dsl_test.go

```diff
@@ -7,7 +7,31 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	pathWithID    = "/path/with/{id}"
+	refPathWithID = "/ref/path/with/id"
+)
+
 func TestBuilder(t *testing.T) {
+	// referenced query param
+	authQ := ogen.NewParameter().
+		InQuery().
+		SetName("auth").
+		SetDescription("Optional bearer token").
+		ToNamedParameter("authInQuery")
+	// referenced header param
+	authH := ogen.NewNamedParameter(
+		"authInHeader",
+		ogen.NewParameter().
+			SetIn("header").
+			SetName("Authorization").
+			SetDescription("Optional bearer token"),
+	)
+	// referenced cookie param
+	csrf := ogen.NewParameter().InCookie().
+		SetName("csrf").
+		SetDescription("CSRF token").ToNamedParameter("csrf")
```

Comment:
Not sure I have the full context, but did you consider using a less-verbose method like:

```go
csrf := ogen.NewParameter().SetIn(...).SetDescription(...).As("csrf")
```

Reply:
Thought about that. But I wanted to make sure that the user knows the last call changes the struct type from `Parameter` to `namedParameter`. 

I'll think about it, maybe I find another solution less ambiguous.